v2functions/blob-trigger-watermark-blob-out-binding/__init__.py

```diff
@@ -0,0 +1,75 @@
+import logging
+import azure.functions as func
+import io
+from PIL import Image
+
+# Final image composite size
+FINAL_COMPOSITE_MAX_HEIGHT = 700
+FINAL_COMPOSITE_MAX_WIDTH = 700
+
+# Set watermark 7 times smaller in width than base image
+WATERMARK_WIDTH_RATIO = 7
+
+# Note the type annotation for our output blob, func.Out[bytes],
+# since it's an image we'll be writing back.
+# See this for more -
+# https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference-python#outputs
+# https://docs.microsoft.com/en-us/python/api/azure-functions/azure.functions.out?view=azure-python
+def main(blobin: func.InputStream, blobout: func.Out[bytes], context: func.Context):
+    logging.info(f"--- Python blob trigger function processed blob \n"
+                 f"----- Name: {blobin.name}\n"
+                 f"----- Blob Size: {blobin.length} bytes")
+    
+    # Pillow calls blobin.read() so only
+    # pass in the image object
+    input_image = blobin
+
+    watermark_image = f'{context.function_directory}/watermark.png'
+
+    base_image = Image.open(input_image)

```

Comment:
It would be better to surround code L29-end with a try/catch block in order to handle NON image files uploaded by users.

Reply:
Agreed. Fix coming soon.

Reply:
How's this?

```
try:
    base_image = Image.open(input_image)
    watermark = Image.open(watermark_image)
except OSError as e:
    print(f'EXCEPTION: Unable to read input as image. {e}')
    sys.exit(255)
except Exception as e:
    print(f'EXCEPTION: {e}')
    sys.exit(255)
```

or should we wrap everything with the `try` block?

Reply:
The try-catch block looks great to me!  Please add it!